gapis/api/vulkan/externs.go

```diff
@@ -108,19 +108,37 @@ func (e externs) onCommandAdded(buffer VkCommandBuffer) {
 	}
 }
 
-func (e externs) enterSubcontext() {
+func (e externs) deferSubmit(ref Submissionʳ) {
 	o := GetState(e.s)
-	o.SubCmdIdx = append(o.SubCmdIdx, 0)
+	o.deferredSubmissions[ref] = append(api.SubCmdIdx{}, o.SubCmdIdx...)
 }
 
-func (e externs) resetSubcontext() {
+func (e externs) executeDeferredSubmit(ref Submissionʳ) {
 	o := GetState(e.s)
-	o.SubCmdIdx = []uint64(nil)
+	o.LastSubCmdIdx = append(api.SubCmdIdx{}, o.SubCmdIdx...)
+	o.SubCmdIdx = append(api.SubCmdIdx{}, o.deferredSubmissions[ref]...)
+}
+
+func (e externs) finishedExecuteDeferredSubmit(ref Submissionʳ) {
+	o := GetState(e.s)
+	o.SubCmdIdx = append(api.SubCmdIdx{}, o.LastSubCmdIdx...)
+	delete(o.deferredSubmissions, ref)
+}
+
+func (e externs) enterSubcontext() {
+	o := GetState(e.s)
+	if len(o.SubCmdIdx) == 0 {
+		o.SubCmdIdx = api.SubCmdIdx{uint64(e.cmdID)}
+	}
+	o.SubCmdIdx = append(o.SubCmdIdx, 0)
```

Comment:
This change breaks the semantics of o.SubCmdIdx: so far, it did not contain the top-level parent command ID, now it does. This is why we have the corresponding change inside MutateSubcommands(), in the sub command index argument passed in the call backs. Now I hope nothing else was relying on o.SubCmdIdx being only the tail of the command indexes.